### PR TITLE
refactoring/TAO-5102/remove_globals_from_core_extension

### PR DESCRIPTION
**Pull request by ionutpad**
- use class constants instead of global constants

### CODE REVIEW

File: controller/DeliveryMgmt.php [hunk 1]
```diff
@@ -115,7 +117,7 @@ public function editDelivery()
         }
         
         // define the groups related to the current delivery
-        $property = new core_kernel_classes_Property(PROPERTY_GROUP_DELVIERY);
+        $property = new core_kernel_classes_Property(GroupAssignment::GROUP_DELIVERY);
```
Comment (antoinerobin) [inline controller/DeliveryMgmt.php]: as it is a property you should keep consistency with other constant name (always put PROP or PROPERTY in the name)

File: model/DeliveryAssemblyService.php [hunk 2]
```diff
@@ -50,7 +57,7 @@ class DeliveryAssemblyService extends \tao_models_classes_ClassService
      */
     public function getRootClass()
     {
-        return new core_kernel_classes_Class(CLASS_COMPILEDDELIVERY);
+        return new core_kernel_classes_Class(DeliveryAssemblyService::CLASS_ID);
```
Comment (antoinerobin) [inline model/DeliveryAssemblyService.php]: you should use self

File: controller/RestDelivery.php [hunk 3]
```diff
@@ -308,7 +309,7 @@ protected function getDeliveryClassByParameters()
      */
     protected function getDeliveryRootClass()
     {
-        return new \core_kernel_classes_Class(CLASS_COMPILEDDELIVERY);
+        return new \core_kernel_classes_Class(DeliveryAssemblyService::CLASS_ID);
```
Comment (antoinerobin) [inline controller/RestDelivery.php]: please call it CLASS_URI

File: controller/DeliveryMgmt.php (hunk 1 above)
Comment (ionutpad) [reply controller/DeliveryMgmt.php]: Ok, good to know, I will change this also in other PR's regarding the same topic

File: controller/RestDelivery.php (hunk 3 above)
Comment (ionutpad) [reply controller/RestDelivery.php]: Noted

**Pull request merged**